docs/navigating-compliance-custom.md

```diff
@@ -0,0 +1,140 @@
+Python importable to Nautobot and the workers. This is a callable function and not a class or other object type.
+
+```python
+PLUGINS_CONFIG = {
+    "nautobot_golden_config": {
+        "get_custom_compliance": "my.custom_compliance.custom_compliance_func"
+    }
+}
+```
+
+## Example
+
+To provide boiler plate code for any future use case, the following is provided
+
+```python
+def custom_compliance_func(obj):
+    # Modify with actual logic, this would always presume compliant.
+    compliance_int = 1
+    compliance = True
+    ordered = True
+    missing = ""
+    extra = ""
+    return {
+        "compliance": compliance,
+        "compliance_int": compliance_int,
+        "ordered": ordered,
+        "missing": missing,
+        "extra": extra,
+    }
+```
+
+Below is an actual example, it takes a very direct approach for matching platform and rule type to a check. This can naturally be modified to apply the abstract logic one may wish to provide.
+
+```python
+# expected_actual_config = '''router bgp 400
+#  no synchronization
+#  bgp log-neighbor-changes
+#  neighbor 70.70.70.70 remote-as 400
+#  neighbor 70.70.70.70 password cisco 
+#  neighbor 70.70.70.70 update-source Loopback80
+#  no auto-summary
+# '''
+import re
+BGP_PATTERN = re.compile("\s*neighbor (?P<ip>\d+\.\d+\.\d+\.\d+) .*")
+BGP_SECRET = re.compile("\s*neighbor (?P<ip>\d+\.\d+\.\d+\.\d+) password (\S+).*")
+def custom_compliance_func(obj):
+    if obj.rule == 'bgp' and obj.device.platform.slug == 'ios':
+        actual_config = obj.actual
+        neighbors = []
+        secrets = []
+        for line in actual_config.splitlines():
+            match = BGP_PATTERN.search(line)
+            if match:
+                neighbors.append(match.groups("ip")[0])
+            secret_match = BGP_SECRET.search(line)
+            if secret_match:
+                secrets.append(match.groups("ip")[0])
+    neighbors = list(set(neighbors))
+    secrets = list(set(secrets))
+    if secrets != neighbors:
+        compliance_int = 0
+        compliance = False
+        ordered = False
+        missing = f"neighbors Found: {str(neighbors)}\nneigbors with secrets found: {str(secrets)}"
+        extra = ""    
+    else:
+        compliance_int = 1
+        compliance = True
+        ordered = True
+        missing = ""
+        extra = ""
+    return {
+        "compliance": compliance,
+        "compliance_int": compliance_int,
+        "ordered": ordered,
+        "missing": missing,
+        "extra": extra,
+    }
+```
+
+In the above example, one may observe that there is no reference to `obj.intended`, that is because this logic is not concerned about such information.
+As the developer of such solutions, you may not require intended configuration or other attributes, but be conscious on the user experience
+implications. It may seem odd to some users to have blank intended configuration but compliance set to true as an example or it may seem odd to have a
```

Comment:
```suggestion
implications. It may seem odd to some users to have blank intended configuration but compliance set to true as an example or it may seem odd to have
```